RIOT-OS/RIOT
# sys/posix/pthread/pthread.c: pthread_create bug fix

### Contribution description


While looking at posix threads I found a bug. When creating more threads than possible `pthread_create` does not check correctly for errors.

### Testing procedure


Try to create more than `MAXTHREADS` pthreads and join on every successfully created thread.
Before the fix it 
Should I implement a test for it?

### Issues/PRs references
